discovery_engine_core/web-api/src/handlers.rs

```diff
@@ -14,33 +14,73 @@
 
 use std::convert::Infallible;
 use warp::hyper::StatusCode;
+use xayn_discovery_engine_ai::CoiSystemState;
+use xayn_discovery_engine_providers::Market;
 
 use crate::{
     db::Db,
-    models::{InteractionRequestBody, UserId},
+    models::{Article, Document, InteractionRequestBody, UserId},
 };
 
-// TODO: TY-3013
-#[allow(clippy::unused_async)]
 pub(crate) async fn handle_ranked_documents(
-    _user_id: UserId,
-    _db: Db,
+    user_id: UserId,
+    db: Db,
 ) -> Result<impl warp::Reply, Infallible> {
-    Ok(StatusCode::NOT_IMPLEMENTED)
+    let mut documents = db
+        .documents
+        .read()
+        .await
+        .clone()
+        .into_values()
+        .collect::<Vec<Document>>();
```

Comment:
It is not very nice to do this every time, we can store a copy of this in the app state.